src/components/app.jsx

```diff
@@ -108,9 +111,11 @@ const App = ({ isPublic }) => {
       <HashRouter>
         <Layout monoColumn={true}>
           {!isPublic && isMobile && (
-            <BarCenter>
-              <AppTitle>{appName}</AppTitle>
-            </BarCenter>
+            <BarContextProvider webviewService={webviewService}>
+              <BarCenter>
+                <AppTitle>{appName}</AppTitle>
+              </BarCenter>
+            </BarContextProvider>
```

Comment:
maybe we could test that change to prevent regression 🙏

Reply:
actually good catch that shouldn't even be there

Reply:
I saw 9 commits to review :( I don't know exactly if i should go into details into the whole code...

Reply:
oh yeah I see, my bad. Indeed this was removed instead of rebased